pkg/util/tracing/grpc_interceptor.go

```diff
@@ -334,6 +343,10 @@ func StreamClientInterceptor(tracer *Tracer, init func(*Span)) grpc.StreamClient
 		streamer grpc.Streamer,
 		opts ...grpc.CallOption,
 	) (grpc.ClientStream, error) {
+		contextWrapper, ok := ctx.(ContextWrapper)
+		if ok {
+			ctx = contextWrapper.UnwrapContext()
```

Comment:
do we always want to unwrap the context? i don't feel like i have enough context into the wrap / unwrap process to stamp here (maybe @knz who reviewed the previous PR?)

Reply:
I think we want it wrapped in 99% of cases so that `Context.Err()` produces a stacktrace (see #95797).

We just need to unwrap it here because gRPC is not looking at the error cause chain https://github.com/grpc/grpc-go/blob/v1.46.0/rpc_util.go#L833.